### PR TITLE
fixed: When deleting a policy an exception thrown in the back

### PR DESCRIPTION
**Pull request by lahiru823**

### CODE REVIEW

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/policy.group.js [hunk 1]
```diff
@@ -387,7 +387,42 @@ $(document).on("click", ".policy-group-delete-button", function () {
 function deletePolicyGroup(applicationId, policyGroupId, policyGroupName) {
```
Comment (lahiru823) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/policy.group.js]: This function has been changed in the latest code

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/edit.asset.js [hunk 2]
```diff
@@ -220,6 +220,21 @@ $(function() {
 
 	$('#editAssetButton').on('click', function() {
 
+		//check if there are any url which doesn't have a policy group
+		var countResourcePolicies = 0;
+		var result = true;
```
Comment (rushmin) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/edit.asset.js]: Variable name doesn't reflect why it is there.

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/edit.asset.js [hunk 3]
```diff
@@ -220,6 +220,21 @@ $(function() {
...
+		$('.policy_groups').each(function () {
+			if ($("#uritemplate_policyGroupId" + countResourcePolicies + " option:selected").text() == "") {
+				result = false;
```
Comment (rushmin) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/edit.asset.js]: Break the loop when this condition is true.

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/logic/create.asset.js [hunk 4]
```diff
@@ -92,6 +92,22 @@ $(function() {
 			return;
 		}
 
+		//check if there are any url which doesn't have a policy group
+		var countResourcePolicies = 0;
+		var result = true;
+		$('.policy_groups').each(function () {
+			if ($("#uritemplate_policyGroupId" + countResourcePolicies + " option:selected").text() == "") {
+				result = false;
```
Comment (rushmin) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/logic/create.asset.js]: Break the loop when this condition is true.

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/logic/create.asset.js [hunk 5]
```diff
@@ -92,6 +92,22 @@ $(function() {
 			return;
 		}
 
+		//check if there are any url which doesn't have a policy group
+		var countResourcePolicies = 0;
+		var result = true;
```
Comment (rushmin) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/logic/create.asset.js]: Variable name doesn't reflect why it is there.

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/policy.group.js [hunk 6]
```diff
@@ -387,7 +387,42 @@ $(document).on("click", ".policy-group-delete-button", function () {
...
+        async: false,
+        url: '/publisher/api/entitlement/policyGroup/associate/url/pattern/list/to/avoid/delete/' + policyGroupId,
+        type: 'GET',
+        contentType: 'application/json',
+        dataType: 'json',
+        success: function (response) {
+            var urlPattern = "";
+            if (response.length != 0) {
+
+                // construct and show the  the warning message with app names which use this partial before delete
```
Comment (rushmin) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/policy.group.js]: Start the comment with a capital letter.

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/edit.asset.js (hunk 2 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/edit.asset.js]: commit# https://github.com/wso2/carbon-appmgt/commit/538c167d27d8e19449f26ffb7210937beb14c76e

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/policy.group.js (hunk 6 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/policy.group.js]: fixed with commit# https://github.com/wso2/carbon-appmgt/commit/3a1d47779e8065a269527864b98d24835d67f89c

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/logic/create.asset.js (hunk 5 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/logic/create.asset.js]: fixed with commit# https://github.com/wso2/carbon-appmgt/commit/9fec8f31af03bf60517c976da761696a0fe2a0eb

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/edit.asset.js (hunk 3 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/edit.asset.js]: fixed with commit# https://github.com/wso2/carbon-appmgt/commit/9a8f198c61a9cde947903790f060b675f8f5f7a5

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/logic/create.asset.js (hunk 4 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/logic/create.asset.js]: fixed with commit# https://github.com/wso2/carbon-appmgt/commit/7e4980d4ceadea25bb2ef10ded84e6e11c17f53b

**Pull request merged**